src/app_neo4j_queries.py

```diff
@@ -693,6 +693,44 @@ def get_children(neo4j_driver, uuid, property_key=None):
     return results
 
 
+def get_source_organ_count(neo4j_driver, uuid: str, organ: str, case_uuid: str = None):
+    """
+    Count the amount of a certain organ is attached to a particular Source.
+
+    Parameters
+    ----------
+    neo4j_driver : neo4j.Driver object
+        The neo4j database connection pool
+    uuid : str
+        The uuid of target entity
+    organ : str
+        The organ to match against
+    case_uuid : str
+        An additional uuid to exclude from the count. Useful during updates.
+
+    Returns
+    -------
+    int
+        The result count
+    """
+    match_case = ''
+    if case_uuid is not None:
+        match_case = f"AND sm.uuid <> '{case_uuid}' "
+
+    query = f"MATCH p=(a:Activity)-[r:USED]->(s:Source) WHERE s.uuid = '{uuid}' " \
```

Comment:
This can be simplified:
MATCH (s:Sample)-[:WAS_GENERATED_BY]->(a)-[:USED]->(sr:Source) where sr.uuid='325e13d30a43386d97fb3d046677b568' and s.organ='BR' return count(s)

Current:
MATCH p=(a:Activity)-[r:USED]->(s:Source) WHERE s.uuid = '325e13d30a43386d97fb3d046677b568' WITH s, COLLECT(a.uuid) as acts MATCH z=(sm:Sample)-[w:WAS_GENERATED_BY]->(a:Activity) WHERE sm.organ='BR' AND a.uuid in acts RETURN length(z) AS result

Also this technically doesn't return the count. It will return an array where the length is the count but each item in the array is `1`

Reply:
Thank you very much @maxsibilla!